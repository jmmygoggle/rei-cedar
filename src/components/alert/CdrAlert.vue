<template>
  <div :class="[$style[baseClass], $style[typeClass]]">
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue';
import propValidator from '../../utils/propValidator';
import { buildClass } from '../../utils/buildClass';

export default defineComponent({
  name: 'CdrAlert',
  props: {
    type: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['info', 'warning', 'success', 'error'],
      ),
      default: 'info',
    },
  },
  setup(props) {
    const baseClass = 'cdr-alert';
    const typeClass = computed(() => props.type && buildClass(baseClass, props.type));
    return {
      baseClass,
      typeClass,
    };
  },
});
</script>

<style lang="scss" module src="./styles/CdrAlert.scss">
</style>
