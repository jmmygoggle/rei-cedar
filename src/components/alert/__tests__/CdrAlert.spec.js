import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrAlert from 'componentdir/alert/CdrAlert';
// TODO: pass array to h function to test proper usage?
// import { h } from 'vue';
// import IconCheckFill from 'componentdir/icon/comps/check-fill';
describe('CdrAlert', () => {
  it('matches snapshot', () => {
    const wrapper = mount(CdrAlert, {
      propsData: {
        type: 'info',
      },
      slots: {
       default: "ðŸ¤ª hey im an alert",
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
