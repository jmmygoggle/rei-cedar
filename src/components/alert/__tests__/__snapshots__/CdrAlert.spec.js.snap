// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrAlert matches snapshot 1`] = `
<div
  class="cdr-alert cdr-alert--info"
>
  
  ðŸ¤ª hey im an alert
  
</div>
`;
